src/app/routes/keycloakSignInRoutes.js

```diff
@@ -0,0 +1,94 @@
+const bodyParser = require('body-parser');
+const envHelper = require('../helpers/environmentVariablesHelper.js');
+const request = require('request-promise');
+var qs = require('querystring');
+var DOMParser = require('xmldom').DOMParser;
+const parser = new DOMParser();
+const { logger } = require('@project-sunbird/logger');
+
+module.exports = (app) => {
+  app.post('/keycloak/login', bodyParser.json(), async (req, res) => {
+    let config = req.body.loginConfig;
+    let formData = {
+      username: req.body.emailId,
```

Comment:
- What if `req` does not contain `body`
- Exception will be thrown cannot read body of undefined; please add exception or validation for input
- All `req.` reference should be handled for exception